galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.manager/src/main/java/dev/galasa/cicsts/internal/CicstsDefaultLogonProvider.java

```diff
@@ -52,10 +65,38 @@ public boolean logonToCicsRegion(ICicsTerminal cicsTerminal) throws CicstsManage
 
             waitForGmText(cicsTerminal);
 
```

Comment:
Have we tested this on both secured and insecure CICS regions?  Is there a major time delay incurred by waiting as we ascertain if we are logging on or not?

Reply:
No, will work out how to do that & test

Reply:
@hobbit1983 No, will work out how to do that, then test

Reply:
@hobbit1983 you were right....switching on all security based SIT parameters and additionally setting `CESL` as the GMTRAN did cause an issue. I've added a `wfk()` which solves it......more importantly though, I've only ever been doing this with `SEC=YES` so need to put in something to cope if a user is specified by `SEC=NO`

Reply:
@hobbit1983 delivered code to account for slow down....if SEC=NO, then added code so it fails fast. I think users can see from console output what's gone wrong. Requesting re-review.